backend/Dockerfile

```diff
@@ -0,0 +1,4 @@
+FROM openjdk:11
+ARG JAR_FILE=target/*.jar
+COPY ${JAR_FILE} app.jar
+ENTRYPOINT ["java","-jar","/app.jar"]
```

Comment:
don't run this as root. use `CMD` instead of `ENTRYPOINT` because it is an application. `ADD` instead of `COPY` ("no magic"). also, order matters -> build layers. [source](https://github.com/bbortt/Buessle-App/blob/develop/Dockerfile):
```
FROM openjdk:11-slim

CMD ["java", "-jar", "haubzit-back.jar"]

RUN useradd -ms /bin/bash haubzit

EXPOSE 8080
EXPOSE 8090
EXPOSE 9092

USER haubzit
WORKDIR /home/haubzit

ARG JAR_FILE
ADD ${JAR_FILE} ./haubzit-back.jar
```

Reply:
It actually was just a basic image to try if it works on gke ^^ But yes I will change some of the mentioned parts. The part with the root user, ports and the part with the layers. However I will probably not change the part with copy and entrypoint unless you re-explain me why we should do this.

COPY OR ADD (https://docs.docker.com/develop/develop-images/dockerfile_best-practices/ -> ADD or COPY):

> Although ADD and COPY are functionally similar, generally speaking, COPY is preferred. That’s because it’s more transparent than ADD. COPY only supports the basic copying of local files into the container, while ADD has some features (like local-only tar extraction and remote URL support) that are not immediately obvious. Consequently, the best use for ADD is local tar file auto-extraction into the image, as in ADD rootfs.tar.xz /.

ENTRYPOINT OR CMD (https://spring.io/guides/topicals/spring-boot-docker): Every official spring guide example uses entrypoint 🤔

Reply:
I can agree on `COPY`.
`CMD` is a suggestions and will often be used with executables. e.g. you can `docker run -it bash echo "hello world"` which is equal to the following Dockerfile:
```
FROM bash
CMD [echo, "hello world"]
```
because the `ENTRYPOINT` in `bash` is the bash itself.
however, we're not providing an executable here, we just want to run that .jar *and only* that .jar. so, I think not allowing anyone to override our default command is nice. well, at least not that easy.